### DISCUSSION TITLE
How can I visualize relationships between entities with displaCy?

### DISCUSSION THREAD

**Discussion by venti07** [explosion/spaCy#11393] [Help: Coding & Implementations] [answered]
How can I visualize relationships between entities with displaCy?
Does anyone know other tools if this is not possible with displaCy?

**Comment by polm** [comment, marked as answer]
displaCy doesn't have a built-in mode for that, but you can just specify the arcs manually - see [this section](https://spacy.io/usage/visualizers#manual-usage) in the docs. 

Here's an example from there of manually specified arcs:

```
{
    "words": [
        {"text": "This", "tag": "DT"},
        {"text": "is", "tag": "VBZ"},
        {"text": "a", "tag": "DT"},
        {"text": "sentence", "tag": "NN"}
    ],
    "arcs": [
        {"start": 0, "end": 1, "label": "nsubj", "dir": "left"},
        {"start": 2, "end": 3, "label": "det", "dir": "left"},
        {"start": 1, "end": 3, "label": "attr", "dir": "right"}
    ]
}
```

In this example the arcs are dependency relations, but you can just modify those to be your relations.

**Reply by venti07** [reply]
Is it also possible to use the style="dep" and style="ent" in combination in one output?

**Reply by polm** [reply]
No, that's not possible. I guess the closest thing you could do would be to use the `"dep"` mode and put the entity labels in the tags.
